### PR TITLE
Fixes to compile in dosbox (using jwasm)

### PR DESCRIPTION
**Pull request by Risca**
Summary of changes:
* fix compatibility with jwasm and masm-6.x, and remove the masm-5.1 compatibility flag
* remove _ASM_SORT.ASM_, which was only small wrapper around `strcmp()`
* remove `TestCDROM()` - it wasn't used.
* rework the linker step so linking can be done with `wmake`. This also removes the need for a special `COPY` command from 4DOS
* rename _MESSAGE_CUSTOM.*_ so it respects the DOS 8.3 file name format

Please see individual commits for details. I'm not an expect in assembly nor ancient assemblers, so please help me find any bugs I may have introduced.

### CODE REVIEW

File: SOURCES/MAKEFILE [hunk 1]
```diff
@@ -50,8 +49,22 @@ OBJETS7  = msg_cust.obj
 
 EXE      = LBA0
 
-$(EXE).exe: $(OBJETS1) $(OBJETS2) $(OBJETS3) $(OBJETS4) $(OBJETS5) $(OBJETS6) $(OBJETS7)
-	echo "run LINK.bat under 4DOS"
+$(EXE).exe: LINK.LD
+	$(WL) Name $@ @ $<
```
Comment (Risca) [inline SOURCES/MAKEFILE:53]: This translates to
```
wlink Name LBA0.exe @ LINK.LD
```
The `@` linker directive means "read more linker directives from this file"

File: LIB386/LIB_SVGA/TEXTURE.ASM [hunk 2]
```diff
@@ -1775,9 +1774,9 @@ h1:		add	bx,4			; next xy
 		add	di,640			; next lig
 
 		dec	boucle
-		jns	#solidloop
+		jns	solidloop
```
Comment (Risca) [inline LIB386/LIB_SVGA/TEXTURE.ASM:1777]: `solidloop` is not defined anywhere! The code still compiles, though :shrug:

File: SOURCES/FIRE.ASM [hunk 3]
```diff
@@ -47,21 +47,21 @@ ComputeFire	proc	uses esi edi ebx edx
...
+		ADD     AL,(1)[eSI]
 		ADC     AH,BH
-		ADD     AL,320[eSI]
+		ADD     AL,(320)[eSI]
 		ADC     AH,BH
-		ADD     AL,319[eSI]
+		ADD     AL,(319)[eSI]
 		ADC     AH,BH
-		ADD     AL,321[eSI]
+		ADD     AL,(321)[eSI]
```
Comment (Risca) [inline SOURCES/FIRE.ASM:64]: I searched the rest of the code for other occurrences of this, but couldn't find any;
```
git grep "[0-9]\["
```

File: SOURCES/MAKEFILE [hunk 4]
```diff
@@ -33,8 +33,7 @@ CFLAGS = /oeaxt /zp2 /4s /zq /v /s /DWATCOM9
 !else
 CFLAGS = /onatx /oh /oi /ei /zp2 /6s /fp6 /s /wx /wcd=131 /wcd=202 /wcd=303 /wcd=308
 !endif
-AFLAGS = /Cx /Zm /c /W0 /Sa /DNoLanguage=SYSCALL /Djumps=; /omf
-LFLAGS = /l=dos4g /x /zq /4s /k7000
```
Comment (Risca) [inline SOURCES/MAKEFILE:37]: These flags are translated to linker directives as follows:
```
/l=dos4g -> system dos4g
/x       -> <removed> (not a linker flag)
/zq      -> <removed> (don't be so quiet)
/4s      -> <removed> (not a linker flag)
/k7000   -> option stack=7000
```

**Comment by xesf** [comment]
Thanks for the PR.
It looks all good but I don't have a way to test this now.

@rmm17 @rsribeiro do you guys have time to help testing this?

File: SOURCES/MAKEFILE [hunk 5]
```diff
@@ -50,8 +49,22 @@ OBJETS7  = msg_cust.obj
...
-	echo "run LINK.bat under 4DOS"
+$(EXE).exe: LINK.LD
+	$(WL) Name $@ @ $<
+
+LINK.LD: $(OBJETS1) $(OBJETS2) $(OBJETS3) $(OBJETS4) $(OBJETS5) $(OBJETS6) $(OBJETS7)
+	if exist $@ del $@
+	echo option stack=7000 >> $@
+	echo system dos4g >> $@
+	for %%G in ($<) do echo File %%G >> $@
+	echo Library %LIB386_PATH%\LIB_3D\LIB_3D.LIB >> $@
```
Comment (rsribeiro) [inline SOURCES/MAKEFILE:60]: This will output absolute paths in the link.ld file. Can we make them relative? Or maybe add link.ld to .gitignore?

**Comment by rsribeiro** [comment]
I was able to compile under windows using open watcom + visual studio assembler and also using open watcom + jwasm. In both cases the game opened and loaded a save game.

jwasm was downloaded from here https://www.japheth.de/JWasm.html#jwdownload

Haven't tested compiling under dosbox yet.

File: SOURCES/MAKEFILE (hunk 5 above)
Comment (Risca) [reply SOURCES/MAKEFILE]: I can do even better than that, I can remove the need for the file altogether :D

File: SOURCES/MAKEFILE (hunk 1 above)
Comment (Risca) [reply SOURCES/MAKEFILE]: Changed in latest diff to having all linker directives passed to the linker without the need for a temporary file

**Comment by rmm17** [comment]
Likewise, I am able to compile and run the game using open watcom + (masm or jwasm). Tested the game until the part where you get the tunic and magic ball, no issues of note found. My understanding of assembly is a bit rusty, but at first glance I haven't found anything outstandingly wrong with the changed code (given the explanations for each change).

**Comment by xesf** [comment]
Thanks @rsribeiro @rmm17, I appreciate.
What version of windows have you test this?

Btw, this remind me that I started some work on bringing both engines together and also porting to masm, but I was not able to continue progress on it. The idea was to be able to compile it under VS2022.

I have made the repo public as it may help you guys: https://github.com/xesf/adeline
(there is a lba1 feature where I started changing some assembly based on what yaz0r did for LBA2).

**Comment by rsribeiro** [comment]
Windows 10 x64

**Comment by rmm17** [comment]
Windows 11 pro x64 here

**Comment by xesf** [comment]
Awesome, thought we were still contraint to some old versions.

**Comment by Risca** [comment]
> Thanks @rsribeiro @rmm17, I appreciate. What version of windows have you test this?

I've only tested this in dosbox, and GitHub workflow here: https://github.com/Risca/lba1-classic-community/actions/runs/4368819976

> Btw, this remind me that I started some work on bringing both engines together and also porting to masm, but I was not able to continue progress on it. The idea was to be able to compile it under VS2022.
> 
> I have made the repo public as it may help you guys: https://github.com/xesf/adeline (there is a lba1 feature where I started changing some assembly based on what yaz0r did for LBA2).

I had this idea about converting the assembly code to C and then look at porting to other platforms, like Linux and possibly ARM :)
Not sure what approach is the easiest, but it sure would be cool to run LBA1 with LBA2 graphics :smiley_cat:

**Comment by Risca** [comment]
There also seems to be a lot of activity for LBA2 over here: https://github.com/sergiou87/lba2-classic-community

**Comment by xesf** [comment]
> 

Thanks, didn't know about it. Following the updates.

**Review by xesf — APPROVED** [review]

**Pull request merged**